### PR TITLE
feat(dockerfile): Add checks for PYTHONHTTPSVERIFY and NODE_TLS_REJECT_UNAUTHORIZED

### PR DESCRIPTION
**Pull request by james-otten-pan**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
Add 2 new checks to detect certain environment variables (`PYTHONHTTPSVERIFY` for Python 2 and `NODE_TLS_REJECT_UNAUTHORIZED` for Node.JS) which, if set to specific values, disable certificate validation. These can be detected in `ENV`, `ARG`, and `RUN` instructions in dockerfiles.

## New/Edited policies
CKV2_DOCKER_5
CKV2_DOCKER_6

### Description
CKV2_DOCKER_5
The `PYTHONHTTPSVERIFY` environment variables disables certificate validation for Python 2.
https://peps.python.org/pep-0493/
https://access.redhat.com/articles/2039753

CKV2_DOCKER_6
The `NODE_TLS_REJECT_UNAUTHORIZED` environment variables disables certificate validation for Node.JS.
https://nodejs.org/api/cli.html#node_tls_reject_unauthorizedvalue

### Fix
Avoid setting the environment variables `PYTHONHTTPSVERIFY` and `NODE_TLS_REJECT_UNAUTHORIZED`. Instead, set up proper certificate validation.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/dockerfile/checks/graph_checks/EnvNodeTlsRejectUnauthorized.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          - RUN
+        attribute: value
+        operator: not_regex_match
+        value: ".*[\\s;&|]+(NODE_TLS_REJECT_UNAUTHORIZED=((0)|('0')|(\"0\"))) .*"
+      - cond_type: attribute
+        resource_types:
+          - RUN
+        attribute: value
+        operator: not_regex_match
+        value: "^(NODE_TLS_REJECT_UNAUTHORIZED=((0)|('0')|(\"0\"))) .*"
```
Comment (gruebel) [inline checkov/dockerfile/checks/graph_checks/EnvNodeTlsRejectUnauthorized.yaml:34]:
```suggestion
    - cond_type: attribute
      resource_types:
        - ARG
        - ENV
      attribute: value
      operator: not_regex_match
      value: "(.*\\s+)?(NODE_TLS_REJECT_UNAUTHORIZED(=|\\s+)0).*"
    - cond_type: attribute
      resource_types:
        - RUN
      attribute: value
      operator: not_regex_match
      value: "(.*[\\s;&|]+)?(NODE_TLS_REJECT_UNAUTHORIZED=((0)|('0')|(\"0\"))) .*"
```
shouldn't this simplify it?

File: checkov/dockerfile/checks/graph_checks/EnvPythonHttpsVerify.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          - RUN
+        attribute: value
+        operator: not_regex_match
+        value: ".*[\\s;&|]+(PYTHONHTTPSVERIFY=((0)|('0')|(\"0\"))) .*"
+      - cond_type: attribute
+        resource_types:
+          - RUN
+        attribute: value
+        operator: not_regex_match
+        value: "^(PYTHONHTTPSVERIFY=((0)|('0')|(\"0\"))) .*"
```
Comment (gruebel) [inline checkov/dockerfile/checks/graph_checks/EnvPythonHttpsVerify.yaml:34]:
```suggestion
    - cond_type: attribute
      resource_types:
        - ARG
        - ENV
      attribute: value
      operator: not_regex_match
      value: "(.*\\s+)?(PYTHONHTTPSVERIFY(=|\\s+)0).*"
    - cond_type: attribute
      resource_types:
        - RUN
      attribute: value
      operator: not_regex_match
      value: "(.*[\\s;&|]+)?(PYTHONHTTPSVERIFY=((0)|('0')|(\"0\"))) .*"
```
same here 🙂

**Review by gruebel — COMMENTED** [review]
nice work, always love the extensive test cases 🏆  I added two suggestions to simplify the checks

File: checkov/dockerfile/checks/graph_checks/EnvNodeTlsRejectUnauthorized.yaml (hunk 1 above)
Comment (james-otten-pan) [reply checkov/dockerfile/checks/graph_checks/EnvNodeTlsRejectUnauthorized.yaml]: Very good call, made that change.

File: checkov/dockerfile/checks/graph_checks/EnvPythonHttpsVerify.yaml (hunk 2 above)
Comment (james-otten-pan) [reply checkov/dockerfile/checks/graph_checks/EnvPythonHttpsVerify.yaml]: Same

File: checkov/dockerfile/checks/graph_checks/EnvNodeTlsRejectUnauthorized.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  category: "APPLICATION_SECURITY"
+definition:
+  or:
+    - cond_type: attribute
+      resource_types:
+        - ARG
+        - ENV
+      attribute: value
+      operator: not_regex_match
+      value: "(.*\\s+)?(NODE_TLS_REJECT_UNAUTHORIZED(=|\\s+)0).*"
```
Comment (james-otten-pan) [inline checkov/dockerfile/checks/graph_checks/EnvNodeTlsRejectUnauthorized.yaml:13]: I looked at this again and did some more testing, more changes to be made here.

**Review by gruebel — APPROVED** [review]
:beers

**Review by ChanochShayner — APPROVED** [review]

**Pull request merged**